CHANGELOG.md

```diff
@@ -10,6 +10,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Bugfixes
 
+* Return only the latest version of each package when a combined index is used. [#849](https://github.com/elastic/package-registry/pull/849)
+* Return only last appearance of the same version of a package when it is available in multiple indexes. [#849](https://github.com/elastic/package-registry/pull/849)
```

Comment:
Let's discuss following scenario:

Developer is working on an integration locally. The package manifest defines the same version as for the embedded package already available in the Docker image, so there are 2 different package revisions published under the same version. Will EPR serve the local mounted package or the Docker embedded one?

Reply:
It depends, if both paths are defined in the same file system indexer, it will return the first one it finds, the second one is [discarded as duplicated](https://github.com/elastic/package-registry/blob/a8303e9bf1bdae76bbe3f87fe2ba1093023352d3/packages/packages.go#L199).

If they are defined in different indexers (plain file system vs zip) in a composable indexer, it will return the last one it finds. This is what is being implemented in this PR.

I can make the composable indexer to return the first package found, to be consistent with a file system indexer with multiple paths. But then I think that I would also reorder the initialization of the zip and plain file system indexers so the zip one takes precedence.

WDYT?

Reply:
The reason why I'm asking is that users will struggle to figure out which package is served when. Maybe it's just better to change the order of indexers to ensure priorities:

1. Local ZIP indexer.
2. Local "unpacked" indexer.
3. Google Storage Indexer.

In the nearest future, I'd like to introduce another indexer, a Proxy indexer?, which will connect to https://epr.elastic.co to pull results from them. This way it won't be necessary to ship EPR as heavy Docker images if the user has access to the internet and can stream relevant images.

Reply:
Done in https://github.com/elastic/package-registry/pull/849/commits/01167538bf3068c015bdc5938beb2d317965a071.